### PR TITLE
v4.1: Check for librt when building LSF support

### PR DESCRIPTION
**Pull request by jjhursey**
bot:notacherrypick

### CODE REVIEW

**Comment by jjhursey** [comment]
Open MPI v4.1.x version of PRRTE https://github.com/openpmix/prrte/pull/809

**Review by awlauria — APPROVED** [review]

File: config/orte_check_lsf.m4 [hunk 1]
```diff
@@ -85,12 +85,21 @@ AC_DEFUN([ORTE_CHECK_LSF],[
                     [orte_check_lsf_happy="no"],
                     [orte_check_lsf_happy="yes"])
 
+          # liblsf requires shm_open, shm_unlink, which are in librt
+          OPAL_SEARCH_LIBS_COMPONENT([shm_open_rt], [shm_open], [rt],
+                        [shm_open_rt_happy="yes"],
+                        [shm_open_rt_happy="no"])
```
Comment (jsquyres) [inline config/orte_check_lsf.m4:91]: Any particular reason you didn't just set `orte_check_lsf_happy` to `yes` or `no`?  I.e., why the additional `AS_IF`?

Comment (jjhursey) [reply config/orte_check_lsf.m4:91]: It matches the `yp_all_nsl_happy` pattern above it. That's the only reason.

Comment (jsquyres) [reply config/orte_check_lsf.m4:91]: If you don't mind, could you consolidate?  No need to propagate a sub-optimal pattern from the past.

Thanks!

Comment (jjhursey) [reply config/orte_check_lsf.m4]: Done. Take a look.

**Review by jsquyres — APPROVED** [review]
I can't test this, but LGTM.

**Pull request merged**